lib/suite.js

```diff
@@ -49,7 +49,7 @@ exports.create = function(parent, title) {
  */
 function Suite(title, parentContext) {
   if (!utils.isString(title)) {
-    throw new Error(
+    throw new TypeError(
```

Comment:
```js
var err = new TypeError(`Suite argument 'title' must be a string. Received type ${typeof title}`);
err.code = 'ERR_INVALID_ARG_TYPE'
throw err;
```

Reply:
Even better...
```js
var err = new TypeError(`Suite argument 'title' must be a string. Received type ${typeof title}`);
err.code = 'ERR_INVALID_ARG_TYPE';
err.argument = 'title';
err.expected = 'string';
err.actual = typeof title;

throw err;
```

Reply:
I'll add this as a new factory for `suite` + `test`, `createArgumentTypeError`.